crates/client/block_import/src/verify_apply.rs

```diff
@@ -0,0 +1,281 @@
+use std::{borrow::Cow, sync::Arc};
+
+use dc_db::{DeoxysBackend, DeoxysStorageError};
+use dp_block::{
+    header::PendingHeader, BlockId, BlockTag, DeoxysBlockInfo, DeoxysBlockInner, DeoxysMaybePendingBlock,
+    DeoxysMaybePendingBlockInfo, DeoxysPendingBlockInfo, Header,
+};
+use dp_convert::ToFelt;
+use starknet_api::core::ChainId;
+use starknet_core::types::Felt;
+use starknet_types_core::hash::{Poseidon, StarkHash};
+
+use crate::{
+    BlockImportError, BlockImportResult, PendingBlockImportResult, PreValidatedBlock, PreValidatedPendingBlock,
+    RayonPool, UnverifiedHeader, ValidatedCommitments, Validation,
+};
+
+mod classes;
+mod contracts;
+
+pub struct VerifyApply {
+    pool: Arc<RayonPool>,
+    backend: Arc<DeoxysBackend>,
+    // Only one thread at once can verify_apply.
```

Comment:
why is that?

Reply:
so, this was the main motivation for separating convertion (now pre-validate) and verify_l2/store_bock (now verify apply) initially:
The global tries necessarily need their updates to be sequential and in order
If a block A changes key K1, the hash of every of its parent nodes need to be updated and recomputed in the global tree, and block B cannot be processed before A is finished because B needs to be aware of that key

The goal of the pre-validate step is to be parallel over blocks, we only compute stuff that donesnt rely on db there to make the verify_apply task lighter. The verify-apply task really is the choke-point then

these names suck by the way if you have better names i'm down to change